src/backend.rs

```diff
@@ -3241,30 +3241,46 @@ impl<'this, M: ModuleContext> Context<'this, M> {
             ValueLocation::Stack(offset) => {
                 let offset = self.adjusted_offset(offset);
                 let temp = self.take_reg(I32).unwrap();
-                let temp_2 = self.take_reg(I32).unwrap();
+                
 
-                dynasm!(self.asm
-                    ; bsr Rd(temp.rq().unwrap()), [rsp + offset]
-                    ; mov Rd(temp_2.rq().unwrap()), DWORD 0x3fu64 as _
-                    ; cmove Rd(temp.rq().unwrap()), Rd(temp_2.rq().unwrap())
-                    ; mov Rd(temp_2.rq().unwrap()), DWORD 0x1fu64 as _
-                    ; xor Rd(temp.rq().unwrap()), Rd(temp_2.rq().unwrap())
-                );
-                self.free_value(ValueLocation::Reg(temp_2));
-                ValueLocation::Reg(temp)
+                if is_x86_feature_detected!("lzcnt") {
```

Comment:
Thanks! This is fine for now but we want to use Cranelift's target system eventually (it's currently not usable outside Cranelift though)